### PR TITLE
Better document what tolerance to use in MappingCartesian.

### PR DESCRIPTION
**Pull request by bangerth**
A follow-up to #14092.

/rebuild

### CODE REVIEW

File: source/fe/mapping_cartesian.cc [hunk 1]
```diff
@@ -55,19 +55,24 @@ is_cartesian(const CellType &cell)
...
   for (const unsigned int v : cell->vertex_indices())
     {
-      const double vertex_tolerance =
-        tolerance * std::max(cell->vertex(v).norm_square(), cell_measure);
-
-      if (cell->vertex(v).distance_square(bounding_box.vertex(v)) >
-          vertex_tolerance)
+      // Choose a tolerance that takes into account both that vertices far
+      // away from the origin have only a finite number of digits
+      // that are considered correct (an "absolute tolerance"), as well as that
```
Comment (kronbichler) [inline source/fe/mapping_cartesian.cc:69]: Would you mind adding the term "cancellation" here, as this is exactly what happens: Far away from the origin, the difference between two vertices is subject to cancellation according to this quantity.

**Comment by bangerth** [comment]
@kronbichler I've updated the comment to mention the issue with cancellation.

As for the magnitude of the tolerance: I think this is correct. We are comparing quantities here in floating point arithmetic. Whether these quantities happen to be the squares of something or not is not important, and so I believe that the tolerances should be on the order of 1e-16 and not 1e-32.

**Review by kronbichler — APPROVED** [review]
OK, let's get this in, thank you.

**Comment by tjhei** [comment]
This is a slight change in functionality as well, so we should cherry pick for 9.4.1.

**Comment by bangerth** [comment]
Anyone have their 9.4 branch still sitting around?

**Pull request merged**